stylelint-polaris/plugins/coverage/index.js

```diff
@@ -2,79 +2,117 @@ const stylelint = require('stylelint');
 
 const {isPlainObject} = require('../../utils');
 
-const ruleName = 'polaris/coverage';
+const coverageRuleName = 'polaris/coverage';
+
+/**
+ * @typedef {import('stylelint').ConfigRules} StylelintRuleConfig
+ * @property {string} message - Message appended to the warning in place of the default category message
+ */
+
+/**
+ * @typedef {object} CategorySettings
+ * @property {import('stylelint').RuleMessage} [message] - Message appended to the warning if no custom message is set on a rule's secondary options
+ * @property {import('stylelint').RuleMeta} [meta] - Category documentation URL hyperlinked to the reported rule in the VS Code diagnostic
+ */
 
 /**
  * @typedef {{
- *   [category: string]: import('stylelint').ConfigRules
- * }} PrimaryOptions
+ *   [category: string]: StylelintRuleConfig | [
+ *     StylelintRuleConfig, CategorySettings
+ *   ]
+ * }} CategorizedRules
  */
 
 // Setting `line` to an invalid line number forces the warning to be reported
 // and the `report({node})` option is used to display the location information:
 // https://github.com/stylelint/stylelint/blob/57cbcd4eb0ee809006a1e3d2ccfe73af48744ad5/lib/utils/report.js#L49-L52
 const forceReport = {line: -1};
+const defaultMeta = {
+  url: 'https://github.com/Shopify/polaris/tree/main/stylelint-polaris/plugins/coverage/README.md#rules',
+};
 
 module.exports = stylelint.createPlugin(
-  ruleName,
-  /** @param {PrimaryOptions} primaryOptions */
-  (primaryOptions, secondaryOptions, context) => {
-    const isPrimaryOptionsValid = validatePrimaryOptions(primaryOptions);
-
-    const rules = [];
-
-    for (const [categoryName, categoryConfigRules] of Object.entries(
-      primaryOptions,
-    )) {
-      for (const [categoryRuleName, categoryRuleSettings] of Object.entries(
-        categoryConfigRules,
+  coverageRuleName,
+  /**
+   * @param {CategorizedRules} categorizedRules - Configured Stylelint rules grouped by Polaris coverage category
+   * @param {import('stylelint').RuleContext} context
+   */
+  (categorizedRules, _, context) => {
+    const isPrimaryOptionsValid = validatePrimaryOptions(categorizedRules);
+    const coverageRules = [];
+
+    for (const [category, categoryConfig] of Object.entries(categorizedRules)) {
+      const [stylelintRules, categorySettings] =
+        normalizeConfig(categoryConfig);
+      for (const [stylelintRuleName, stylelintRuleConfig] of Object.entries(
+        stylelintRules,
       )) {
-        rules.push({
-          coverageRuleName: `polaris/${categoryName}/${categoryRuleName}`,
-          categoryRuleName,
-          categoryRuleSettings,
-          categoryRuleSeverity: categoryRuleSettings?.[1]?.severity,
-          categoryRuleFix:
-            context.fix && !categoryRuleSettings?.[1]?.disableFix,
+        const ruleName = `polaris/${category}/${deNamespaceRuleName(
```

Comment:
**question:** Why are we removing the namespace? This is how custom rules are distinguished from the Stylelint built-ins (e.g. [polaris/at-rule-disallowed-list](https://github.com/Shopify/polaris/tree/main/stylelint-polaris/plugins/at-rule-disallowed-list) vs Stylelint's [at-rule-disallowed-list](https://stylelint.io/user-guide/rules/at-rule-disallowed-list/))

Reply:
Reverting as it looks like the disable comments won't work if the duplicate namespace is removed, as the disables are based on what rules are enabled not on what rules are reported 😢

Reply:
Gotcha, yea I have a slight preference to keep namespaces to continue Stylelint's conventions of disambiguating built-in rules from custom rules. Also, to avoid (future) rule name conflicts and ensuring a `stylelint-disable` comment like the following is possible:
```js
// stylelint-disable polaris/colors/at-rule-disallowed-list, polaris/colors/polaris/at-rule-disallowed-list
```

Totally, understanding wanting to dedupe `polaris` though

Reply:
Ah yeah, that makes total sense 💯